cpmpy/solvers/gcs.py

```diff
@@ -0,0 +1,413 @@
+            (technical side note: any constraints created during conversion of the objective
+            are permanently posted to the solver)
+        """
+        # make objective function non-nested
+        (flat_obj, flat_cons) = flatten_objective(expr)
+        self += flat_cons # add potentially created constraints
+        self.user_vars.update(get_variables(flat_obj)) # add objvars to vars
+
+        # make objective function or variable and post
+        obj = self._make_numexpr(flat_obj)
+        self.has_objective = True
+        self.objective_var = obj
+        if minimize:
+            self.gcs.minimise(obj)  
+        else:
+            self.gcs.maximise(obj)
+
+    def _make_numexpr(self, cpm_expr):
+        """
+            Turns a numeric CPMpy 'flat' expression into a solver-specific
+            numeric expression
+            Used especially to post an expression as objective function
+        """
+        if is_num(cpm_expr):
+            return cpm_expr
+
+        # decision variables, check in varmap
+        if isinstance(cpm_expr, _NumVarImpl):  # _BoolVarImpl is subclass of _NumVarImpl
+            return self.solver_var(cpm_expr)
+
+        (obj_var, obj_cons) = get_or_make_var(cpm_expr)
+        self += obj_cons
+        return self.solver_var(obj_var)
+
+    def transform(self, cpm_expr):
+        """
+            Transform arbitrary CPMpy expressions to constraints the solver supports
+
+            Implemented through chaining multiple solver-independent **transformation functions** from
+            the `cpmpy/transformations/` directory.
+
+            See the 'Adding a new solver' docs on readthedocs for more information.
+
+        :param cpm_expr: CPMpy expression, or list thereof
+        :type cpm_expr: Expression or list of Expression
+
+        :return: list of Expression
+        """
+        # apply transformations, then post internally
+        # expressions have to be linearized to fit in MIP model. See /transformations/linearize
+        cpm_cons = toplevel_list(cpm_expr)
+        supported = {
+            "min", 
+            "max", 
+            "abs", 
+            "alldifferent", 
+            "element", 
+            'table', 
+            'negative_table', 
+            'count', 
+            'nvalue',
+            'inverse', 
+            'circuit', 
+            'xor'}
+        cpm_cons = decompose_in_tree(cpm_cons, supported)
+        cpm_cons = flatten_constraint(cpm_cons)  # flat normal form
+
+        # NB: GCS supports full reification for linear equality and linear inequaltiy constraints
+        # but no reification for linear not equals and not half reification for linear equality. 
+        # Maybe a future transformation (or future work on the GCS solver).
+        cpm_cons = reify_rewrite(cpm_cons, supported=frozenset(['==']))
+        cpm_cons = only_numexpr_equality(cpm_cons, supported=frozenset(["sum", "wsum"]))  # supports >, <, !=
```

Comment:
If you want some of your reified sum/wsums to pass, you could not reify_rewrite them above and copy the only_numexpr_equality function (which checks for reification, the documentation is outdated) and handle the special cases there..